### PR TITLE
chore: fix post-upgrade job when it is set to null

### PR DESCRIPTION
**Pull request by alejandroEsc**
If a cert is null, this should still process and work. Here we add checks against that possibility.

### CODE REVIEW

File: charts/redpanda/templates/post-upgrade.yaml [hunk 1]
```diff
@@ -91,7 +91,7 @@ spec:
             if [ -d "/etc/secrets/users/" ]; then
                 IFS=":" read -r USER_NAME PASSWORD MECHANISM < <(grep "" $(find /etc/secrets/users/* -print))
                 curl -svm3 --fail --retry "120" --retry-max-time "120" --retry-all-errors --ssl-reqd \
-  {{- if $cert.caEnabled }}
+  {{- if and $cert ( dig "caEnabled" false $cert ) }}
```
Comment (RafalKorepta) [inline charts/redpanda/templates/post-upgrade.yaml:94]: That problem repeats in many files

- https://github.com/redpanda-data/helm-charts/blob/9c4c5854a6deb22558d6dca25ccb485ea2b15eb8/charts/redpanda/templates/tests/test-pandaproxy-internal-tls-status.yaml#L65
- https://github.com/redpanda-data/helm-charts/blob/9c4c5854a6deb22558d6dca25ccb485ea2b15eb8/charts/redpanda/templates/tests/test-pandaproxy-internal-tls-status.yaml#L74
- https://github.com/redpanda-data/helm-charts/blob/9c4c5854a6deb22558d6dca25ccb485ea2b15eb8/charts/redpanda/templates/tests/test-schemaregistry-internal-tls-status.yaml#L96
- https://github.com/redpanda-data/helm-charts/blob/9c4c5854a6deb22558d6dca25ccb485ea2b15eb8/charts/redpanda/templates/tests/test-schemaregistry-internal-tls-status.yaml#L107
- https://github.com/redpanda-data/helm-charts/blob/9c4c5854a6deb22558d6dca25ccb485ea2b15eb8/charts/redpanda/templates/tests/test-schemaregistry-internal-tls-status.yaml#L123
- https://github.com/redpanda-data/helm-charts/blob/9c4c5854a6deb22558d6dca25ccb485ea2b15eb8/charts/redpanda/templates/tests/test-schemaregistry-internal-tls-status.yaml#L139
- https://github.com/redpanda-data/helm-charts/blob/9c4c5854a6deb22558d6dca25ccb485ea2b15eb8/charts/redpanda/templates/tests/test-schemaregistry-internal-tls-status.yaml#L149
- https://github.com/redpanda-data/helm-charts/blob/9c4c5854a6deb22558d6dca25ccb485ea2b15eb8/charts/redpanda/templates/_configmap.tpl#L600
- https://github.com/redpanda-data/helm-charts/blob/9c4c5854a6deb22558d6dca25ccb485ea2b15eb8/charts/redpanda/templates/_configmap.tpl#L613

**Comment by RafalKorepta** [comment]
> chore: fix post-upgrade job when it is set to null

NIT: It's not chore change in my opinion as @vuldin needs a Redpanda deployment that can be achieved without cert manager. 

I would like to suggest that any occurrence of

```
{{- $cert := get .Values.tls.certs .Values.listeners.kafka.tls.cert }}
```

is a place to address entire problem.

**Comment by alejandroEsc** [comment]
> > chore: fix post-upgrade job when it is set to null
> 
> NIT: It's not chore change in my opinion as @vuldin needs a Redpanda deployment that can be achieved without cert manager.
> 
> I would like to suggest that any occurrence of
> 
> ```
> {{- $cert := get .Values.tls.certs .Values.listeners.kafka.tls.cert }}
> ```
> 
> is a place to address entire problem.

🤔 but the null issue only appears in the operator. If i understand correctly, you are ok with the solution, but you want to be holistic and take care of each instance. Could I open an issue and do so for your findings in another PR?

**Review by RafalKorepta — APPROVED** [review]
I retested and it is working fine. I'm not sure why other occurrences mentioned in my comment are not complaining.

**Pull request merged**